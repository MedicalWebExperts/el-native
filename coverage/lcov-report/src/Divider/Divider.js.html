<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/Divider/Divider.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/Divider</a> Divider.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>9/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/9</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { string, oneOf } from 'prop-types';
import { StyleSheet, View } from 'react-native';
&nbsp;
// TODO: Get theme colors
const colors = {
  primary: 'lightgreen',
  background: 'lightgray',
};
&nbsp;
const lineType = {
  none: '0%',
  short: '20%',
  medium: '40%',
  long: '60%',
};
&nbsp;
const styles = StyleSheet.create({
  divider: {
    height: 3,
    backgroundColor: colors.primary,
  },
  dividerWrapper: {
    width: '100%',
    height: 0.5,
    backgroundColor: colors.background,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
&nbsp;
const propTypes = {
  /**
   * Color of the long thin line divider. Default: Theme primary color.
   */
  color: string,
  /**
   * Color of the short thick line divider. Default: Theme light gray color.
   */
  primaryColor: string,
  /**
   * Length of the short thick line divider. Default: none.
   */
  length: oneOf(['short', 'medium', 'long', 'none']),
};
const defaultProps = {
  color: colors.brand,
  primaryColor: null,
  length: 'none',
};
&nbsp;
/**
 * Divider renders a thin View to separate content.
 */
const Divider = props =&gt; (
  &lt;View style={[styles.dividerWrapper, props.color &amp;&amp; { backgroundColor: props.color }]}&gt;
    &lt;View
      style={[
        styles.divider,
        props.primaryColor &amp;&amp; { backgroundColor: props.primaryColor },
        { width: lineType[props.length] },
      ]}
    /&gt;
  &lt;/View&gt;
);
&nbsp;
Divider.propTypes = propTypes;
Divider.defaultProps = defaultProps;
&nbsp;
export default Divider;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jan 17 2018 12:58:56 GMT-0300 (-03)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
