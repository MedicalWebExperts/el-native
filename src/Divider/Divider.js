import React from 'react';
import { string, oneOf } from 'prop-types';
import { StyleSheet, View } from 'react-native';

const accentLength = {
  none: '0%',
  short: '20%',
  medium: '40%',
  long: '60%',
};

const styles = StyleSheet.create({
  divider: {
    height: 3,
    backgroundColor: 'grey',
  },
  dividerWrapper: {
    width: '100%',
    height: 0.5,
    backgroundColor: 'grey',
    justifyContent: 'center',
    alignItems: 'center',
  },
});

const propTypes = {
  /**
   * Color of the divider. Default: Theme light gray color.
   */
  color: string,
  /**
   * Color of the short thick line divider. Default: Theme primary color.
   */
  primaryColor: string,
  /**
   * Length of the short thick accent. Default: none.
   */
  accentLength: oneOf(['short', 'medium', 'long', 'none']),
};
const defaultProps = {
  color: 'grey',
  primaryColor: null,
  accentLength: 'none',
};

/**
 * Divider renders a thin View to separate content.
 */
const Divider = props => (
  <View style={[styles.dividerWrapper, props.color && { backgroundColor: props.color }]}>
    <View
      style={[
        styles.divider,
        props.primaryColor && { backgroundColor: props.primaryColor },
        { width: accentLength[props.accentLength] },
      ]}
    />
  </View>
);

Divider.propTypes = propTypes;
Divider.defaultProps = defaultProps;

export default Divider;
