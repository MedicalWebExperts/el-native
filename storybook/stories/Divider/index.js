import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { StyleSheet, View, Dimensions } from 'react-native';

import CenterView from '../CenterView';
import { Divider } from '../../../src';

const { width } = Dimensions.get('window');

const styles = StyleSheet.create({
  wrapper: {
    justifyContent: 'space-between',
    alignItems: 'center',
    width,
    height: 240,
  },
});

const customColor = 'darkgray';
const customPrimaryColor = 'aquamarine';

storiesOf('Divider', module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('Default', () => (
    <View style={styles.wrapper}>
      <Divider />
      <Divider length="short" />
      <Divider length="medium" />
      <Divider length="long" />
    </View>
  ))
  .add('Custom color', () => (
    <View style={styles.wrapper}>
      <Divider color={customColor} />
      <Divider length="short" color={customColor} />
      <Divider length="medium" color={customColor} />
      <Divider length="long" color={customColor} />
    </View>
  ))
  .add('Custom primary color', () => (
    <View style={styles.wrapper}>
      <Divider primaryColor={customPrimaryColor} />
      <Divider length="short" primaryColor={customPrimaryColor} />
      <Divider length="medium" primaryColor={customPrimaryColor} />
      <Divider length="long" primaryColor={customPrimaryColor} />
    </View>
  ))
  .add('Custom colors', () => (
    <View style={styles.wrapper}>
      <Divider color={customColor} primaryColor={customPrimaryColor} />
      <Divider length="short" color={customColor} primaryColor={customPrimaryColor} />
      <Divider length="medium" color={customColor} primaryColor={customPrimaryColor} />
      <Divider length="long" color={customColor} primaryColor={customPrimaryColor} />
    </View>
  ));
