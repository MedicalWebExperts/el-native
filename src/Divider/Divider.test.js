import React from 'react';
import renderer from 'react-test-renderer';

import Divider from './Divider';

const customColor = 'darkgray';
const customPrimaryColor = 'aquamarine';

describe('Divider Snapshots', () => {
  it('should render a simple Divider ', () => {
    const component = renderer.create(<Divider />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a short Divider', () => {
    const component = renderer.create(<Divider length="short" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a medium Divider', () => {
    const component = renderer.create(<Divider length="medium" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a long Divider', () => {
    const component = renderer.create(<Divider length="long" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a Divider with custom colors', () => {
    const component = renderer.create(
      <Divider color={customColor} primaryColor={customPrimaryColor} />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
