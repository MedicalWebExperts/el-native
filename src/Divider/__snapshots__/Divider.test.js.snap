// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Snapshots should render a Divider with custom colors 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "height": 0.5,
        "justifyContent": "center",
        "width": "100%",
      },
      Object {
        "backgroundColor": "darkgray",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "lightgreen",
          "height": 3,
        },
        Object {
          "backgroundColor": "aquamarine",
        },
        Object {
          "width": "0%",
        },
      ]
    }
  />
</View>
`;

exports[`Divider Snapshots should render a long Divider 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "height": 0.5,
        "justifyContent": "center",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "lightgreen",
          "height": 3,
        },
        null,
        Object {
          "width": "60%",
        },
      ]
    }
  />
</View>
`;

exports[`Divider Snapshots should render a medium Divider 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "height": 0.5,
        "justifyContent": "center",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "lightgreen",
          "height": 3,
        },
        null,
        Object {
          "width": "40%",
        },
      ]
    }
  />
</View>
`;

exports[`Divider Snapshots should render a short Divider 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "height": 0.5,
        "justifyContent": "center",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "lightgreen",
          "height": 3,
        },
        null,
        Object {
          "width": "20%",
        },
      ]
    }
  />
</View>
`;

exports[`Divider Snapshots should render a simple Divider  1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "height": 0.5,
        "justifyContent": "center",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "lightgreen",
          "height": 3,
        },
        null,
        Object {
          "width": "0%",
        },
      ]
    }
  />
</View>
`;
